### DISCUSSION TITLE
How to Automatically Replace Specific Characters in InputQueryText?

### DISCUSSION THREAD

**Discussion by shlee-gsp** [microsoft-search/pnp-modern-search#4126] [Q&A] [answered]
Hello,

I’m a beginner developer working with SharePoint, and I’ve encountered an issue while building a site using the PnP Modern Search library. I’m using the PnP Search Results web part, where I’ve configured the Query Template with a line like:
**Title: *{inputQueryText}***

This setup allows users to search for documents within the site by entering a keyword in the search bar. However, I’ve noticed a problem: many of the document titles in my library include hyphens (-), such as AA-11-BBB.

For example, when a user searches for AA-11, the document is not returned in the search results, even though it clearly exists. After testing with the SharePoint Search API, I found that replacing the hyphen (-) in the query with an asterisk (*) allows the search to function as intended (e.g., searching for AA*11 successfully returns the desired document).

My goal is to automatically replace any hyphen (-) in the user’s input (inputQueryText) with an asterisk (*) before the query is executed. However, I haven’t been able to find a way to achieve this within the PnP Modern Search library or documentation.

Could you please advise if this functionality is possible? If so, I would greatly appreciate any guidance or examples on how to implement it.

Thank you for your support!

Best regards,
Kyle

**Comment by kasperbolarsen** [comment, marked as answer]
Hyphens is the NOT operarator in the Keyword Query Language which is used in search, and your issue is very well known, and by no means restricted to PnP Modern Search, Microsoft Search in the top of your sites is using the same query language. In most cases this issue is handled by training people to use parentheses in their query,  "AA-11-BBB" will return the items where that specific text block/word is found.
In some custom search portal we also adds reminders in the text of the seach box or in a text web part on the page
![image](https://github.com/user-attachments/assets/4a04f163-2f80-4b5d-b8f2-e9a6fe2278d5)
